src/main/java/de/tum/in/www1/artemis/web/rest/StaticCodeAnalysisResource.java

```diff
@@ -0,0 +1,146 @@
+     *
+     * @param exerciseId of the the exercise
+     * @param categories used for the update
+     * @return the updated static code analysis categories
+     */
+    @PatchMapping(Endpoints.CATEGORIES)
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Set<StaticCodeAnalysisCategory>> updateStaticCodeAnalysisCategories(@PathVariable Long exerciseId,
+            @RequestBody Set<StaticCodeAnalysisCategory> categories) {
+        log.debug("REST request to update static code analysis categories for programming exercise {}", exerciseId);
+
+        ProgrammingExercise programmingExercise = programmingExerciseService.findById(exerciseId);
+
+        if (!Boolean.TRUE.equals(programmingExercise.isStaticCodeAnalysisEnabled())) {
+            return badRequest();
+        }
+
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(programmingExercise)) {
+            return forbidden();
+        }
+
+        // Validate the category updates
+        for (var category : categories) {
+            // Each categories must have an id
+            if (category.getId() == null) {
+                return badRequest(ENTITY_NAME, "scaCategoryIdError", "Static code analysis category id is missing.");
+            }
+
+            // Penalty must not be null or negative
+            if (category.getPenalty() == null || category.getPenalty() < 0) {
+                return badRequest(ENTITY_NAME + " " + category.getId(), "scaCategoryPenaltyError",
+                        "Penalty for static code analysis category " + category.getId() + " must be a non-negative integer.");
+            }
+
+            // MaxPenalty must not be smaller than penalty
+            if (category.getMaxPenalty() != null && category.getPenalty() > category.getMaxPenalty()) {
+                return badRequest(ENTITY_NAME + " " + category.getId(), "scaCategoryMaxPenaltyError",
+                        "Max Penalty for static code analysis category " + category.getId() + " must not be smaller than the penalty.");
+            }
+
+            // Category state must not be null
+            if (category.getState() == null) {
+                return badRequest(ENTITY_NAME + " " + category.getId(), "scaCategoryStateError",
+                        "Max Penalty for static code analysis category " + category.getId() + " must not be smaller than the penalty.");
+            }
+
+            // Exercise id of the request path must match the exerciseId in the request body if present
+            if (category.getExercise() != null && !Objects.equals(category.getExercise().getId(), exerciseId)) {
+                return conflict(ENTITY_NAME + " " + category.getId(), "scaCategoryExerciseIdError",
+                        "Exercise id path variable does not match exercise id of static code analysis category " + category.getId());
+            }
```

Comment:
Maybe you can refactor this logic into a different method, it makes this method quite large and I guess it might be useful elsewhere.

Reply:
Created a own method for the validation.